src/extension.ts

```diff
@@ -69,6 +71,9 @@ export async function activate(
   if (context.extensionMode !== vscode.ExtensionMode.Test) {
     handlePluginInstalled(context);
   }
+  const statePoller = new StatePoller();
```

Comment:
state poller can be global to all objects polling state. that was the point of having the statics

Reply:
if going this route you need to pass the poller to all the relevant places when that will be implemented (I didn't touch any of the existing pollers because of time constraints)